src/reflect/scala/reflect/io/ZipArchive.scala

```diff
@@ -110,15 +129,17 @@ abstract class ZipArchive(override val file: JFile, release: Option[String]) ext
     //   parent.entries(baseName(path)) = dir
     //   dir
     // })
-    dirs get path match {
+    val dotted = pathToDotted(path)
+    dirs get dotted match {
```

Comment:
It seem that `dirs` here can also be `ManifestResources.dirs` and `URLZipArchive.dirs`, so these maps need to use dotted strings as keys too.